engine/runtime/src/main/java/org/enso/interpreter/runtime/error/DataflowError.java

```diff
@@ -49,16 +50,19 @@ public final class DataflowError extends AbstractTruffleException implements Ens
    *
    * @param payload the user-provided value carried by the error
    * @param location the node in which the error was created
+   * @param attachFullStackTraceProfile
    * @return a new dataflow error
    */
-  public static DataflowError withDefaultTrace(State state, Object payload, Node location) {
+  public static DataflowError withDefaultTrace(
+      State state, Object payload, Node location, BranchProfile attachFullStackTraceProfile) {
     assert payload != null;
     boolean attachFullStackTrace =
         state == null
             || EnsoContext.get(location)
                 .getExecutionEnvironment()
                 .hasContextEnabled("Dataflow_Stack_Trace");
     if (attachFullStackTrace) {
+      attachFullStackTraceProfile.enter();
```

Comment:
This line fixes the problem? That is surprising:
- I can imagine it makes the compilation graph way smaller.
- I have no idea why it should make anything faster?

Are we getting bailout on compilation? Is the compilation still in progress when benchmarking?

Reply:
Outdated - the implementation is now different.